hummingbot/strategy/strategy_base.pyx

```diff
@@ -48,6 +48,14 @@ cdef class FundingPaymentCompletedListener(BaseStrategyEventListener):
         self._owner.c_did_complete_funding_payment(arg)
 
 
+cdef class PositionModeChangeSuccessListener(BaseStrategyEventListener):
+    cdef c_call(self, object arg):
+        self._owner.c_did_change_position_mode_succeed(arg)
+
+cdef class PositionModeChangeFailureListener(BaseStrategyEventListener):
+    cdef c_call(self, object arg):
+        self._owner.c_did_change_position_mode_fail(arg)
+
```

Comment:
How will it be then received by the strategy? The strategy needs to take decisions and inform the user in case of a failure. As per previous discussions, the client should not cancel outstanding orders or close positions automatically.